### PR TITLE
Disable Webhooks

### PR DESCRIPTION
**Pull request by bshephar**
This change builds the Operator bundle with ENABLE_WEBHOOKS=false. This is due to the duplication of webhook logic since openstack-operator imports and runs this webhook at the OpenStackControlPlane level. Therefor, having the webhook run by the service operator is a needless duplication of this webhook logic.

Jira: https://issues.redhat.com/browse/OSPRH-11198
Depends-On: https://github.com/openstack-k8s-operators/openstack-k8s-operators-ci/pull/114

### CODE REVIEW

**Comment by bshephar** [comment]
Yeah, so as we discussed on the call. This will break Kuttl tests. So we need to be able to dynamically set this for CI jobs - which is ideal since we _do_ want to test web hooks in CI. But, ultimately, the bundle we build and ship should have the service level operator web hooks disabled so that they can be executed by the openstack-operator without duplication.

**Comment by bshephar** [comment]
We need to merge this one to proceed any further here:
https://github.com/openshift/release/pull/58391

**Comment by bshephar** [comment]
So, the build is enabling webhooks:
```
❯ curl -s "https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openstack-k8s-operators_heat-operator/466/pull-ci-openstack-k8s-operators-heat-operator-main-heat-operator-build-deploy-kuttl/1859192010674540544/artifacts/heat-operator-build-deploy-kuttl/openstack-k8s-operators-build/build-log.txt" | grep ENABLE_WEB
[2/2] STEP 10/21: ARG ENABLE_WEBHOOKS=false
[2/2] STEP 18/21: ENV ENABLE_WEBHOOKS="${ENABLE_WEBHOOKS}"
+ oc patch bc/heat-operator-bundle --type=json '-p=[{"op": "add", "path": "/spec/strategy/dockerStrategy/env", "value": [{"name": "ENABLE_WEBHOOKS", "value": "true"}]}]'
STEP 2/17: ENV "ENABLE_WEBHOOKS"="true"
+ oc patch bc/openstack-operator-bundle --type=json '-p=[{"op": "add", "path": "/spec/strategy/dockerStrategy/env", "value": [{"name": "ENABLE_WEBHOOKS", "value": "true"}]}]'
STEP 2/17: ENV "ENABLE_WEBHOOKS"="true"
```

But Kuttl is failing because the webhook isn't responding:
```
❯ curl -s "https://storage.googleapis.com/test-platform-results/pr-logs/pull/openstack-k8s-operators_heat-operator/466/pull-ci-openstack-k8s-operators-heat-operator-main-heat-operator-build-deploy-kuttl/1859192010674540544/build-log.txt" | grep -E 'vheat|mheat'
    case.go:380: Internal error occurred: failed calling webhook "mheat.kb.io": failed to call webhook: Post "https://heat-operator-controller-manager-service.openstack-operators.svc:443/mutate-heat-openstack-org-v1beta1-heat?timeout=10s": dial tcp 10.128.0.98:9443: connect: connection refused
    case.go:380: Internal error occurred: failed calling webhook "mheat.kb.io": failed to call webhook: Post "https://heat-operator-controller-manager-service.openstack-operators.svc:443/mutate-heat-openstack-org-v1beta1-heat?timeout=10s": dial tcp 10.128.0.98:9443: connect: connection refused
    case.go:380: Internal error occurred: failed calling webhook "mheat.kb.io": failed to call webhook: Post "https://heat-operator-controller-manager-service.openstack-operators.svc:443/mutate-heat-openstack-org-v1beta1-heat?timeout=10s": dial tcp 10.128.0.98:9443: connect: connection refused
```

I must be missing something else.

**Comment by bshephar** [comment]
... Yeah, ARG doesn't work like that.

File: Makefile [hunk 1]
```diff
@@ -60,6 +60,8 @@ ENVTEST_K8S_VERSION = 1.29
 # Set minimum Go version
 GOTOOLCHAIN_VERSION ?= go1.21.0
 
+ENABLE_WEBHOOKS ?= true
```
Comment (rabi) [inline Makefile:63]: This default is set to true? Where do we set it to  false? Is this about not including the webhook definitions in the bundle or not registering them with the controller manager?

**Comment by rabi** [comment]
You can still see the webhook definitions in the csv  and webhooks registered with the controller manager 
[1] https://logserver.rdoproject.org/66/466/2e89b036df8fb622dd793ee602363754d6849630/github-check/heat-operator-tempest-multinode/32f4769/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/csv/heat-operator.v0.0.1
[2] https://logserver.rdoproject.org/66/466/2e89b036df8fb622dd793ee602363754d6849630/github-check/heat-operator-tempest-multinode/32f4769/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/namespaces/openstack-operators/pods/heat-operator-controller-manager-7574fb7f67-cpz4h/logs/manager.log

**Comment by dprince** [comment]
@bshephar So my PR here would disable the validation/mutating webhooks for operators here: https://github.com/openstack-k8s-operators/openstack-operator/pull/1185

**Comment by bshephar** [comment]
> @bshephar So my PR here would disable the validation/mutating webhooks for operators here: [openstack-k8s-operators/openstack-operator#1185](https://github.com/openstack-k8s-operators/openstack-operator/pull/1185)

We wanted to run them from the openstack-operator though didn't we? Then disable them in each individual service operator.

**Comment by bshephar** [comment]
This doesn't need to be done on an individual operator level. We will change the architecture for deploying openstack-operator and in doing so, we will stop running service webhooks.

**Pull request closed without merging**